source/common/http/utility.h

```diff
@@ -676,6 +676,35 @@ getMergedPerFilterConfig(const Http::StreamFilterCallbacks* callbacks,
   return merged;
 }
 
+/**
+ * Return all the available per route filter configs
+ *
+ * @param callbacks The stream filter callbacks to check for route configs.
+ *
+ * @return The all available per route config. The returned pointers are guaranteed to be non-null
+ * and their lifetime is the same as the matched route.
+ */
+template <class ConfigType>
+std::vector<const ConfigType*> getAllPerFilterConfig(const Http::StreamFilterCallbacks* callbacks) {
+  static_assert(std::is_copy_constructible<ConfigType>::value,
+                "ConfigType must be copy constructible");
+  ASSERT(callbacks != nullptr);
```

Comment:
Prefer `absl::InlinedVector` here to avoid unnecessary heap allocation.
And why this static_assert is necessary?

Reply:
Sure! I will switch to `absl::InlinedVector`

assert is not necessary, but the whole point of assert IMO is sanity/safe check rather than necessity. It mostly from forward thinking perspective that if implementation changes in the future, we want to assert non-null before accessing the pointer.  I am OK if you want me remove it.

Reply:
It's OK to add additional guard to the code. But please at least to add a comment to it.